packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -394,7 +417,7 @@ export class Tracing implements InstrumentationListener, SnapshotterDelegate, Ha
     let error: Error | undefined;
     let result: T | undefined;
     this._writeChain = this._writeChain.then(async () => {
-      if (!this._context._browser.isConnected())
+      if (this._context && !this._context._browser.isConnected())
```

Comment:
While you are here, could you please add a comment that "this check is here because closing the browser removes the tracesDir and tracing dies trying to archive". If we end up reusing tracesDir between multiple APIRequestContext tracers, there might be a similar issue I guess.

Reply:
Done.